app/services/category_hashtag_data_source.rb

```diff
@@ -48,15 +50,38 @@ def self.search(guardian, term, limit)
         category[:name].downcase.include?(term) || category[:slug].downcase.include?(term)
       end
       .take(limit)
-      .map { |category| category_to_hashtag_item(guardian_categories, category) }
+      .map do |category|
+        parent_category =
+          guardian_categories.find { |cat| cat[:id] == category[:parent_category_id] }
+        category_to_hashtag_item(parent_category, category)
+      end
   end
 
   def self.search_sort(search_results, term)
-    search_results
-      .select { |item| item.slug == term }
-      .sort_by { |item| item.text.downcase }
-      .concat(
-        search_results.select { |item| item.slug != term }.sort_by { |item| item.text.downcase },
+    if term.present?
+      search_results.sort_by { |item| [item.slug == term ? 0 : 1, item.text.downcase] }
+    else
+      search_results.sort_by { |item| item.text.downcase }
+    end
+  end
+
+  def self.search_without_term(guardian, limit)
+    Category
```

Comment:
I learnt of this while working on https://github.com/discourse/discourse/commit/207b764ea37f13469db1117c5c1102ace05fd2f9 but I'm wondering if we should exclude categories with `Category#search_priority` set to `ignore`.